[4차 스프린트 - Review] 내가 받은 리뷰 조회, 비공개 설정 기능 구현/Review 관련 Controller 테스트 및 문서화

na-yk opened the pull request:
## ✅ PR 체크리스트
- [x] **테스트**
- [x] **문서화 작업**
## 💡 어떤 작업을 하셨나요?
**Issue Number**
close #129
close #128 

**작업 내용**
- [x] 리뷰 생성 기능 Controller 테스트 및 문서화
- [x] 리뷰 코멘트 공개 여부 변경 기능 구현
- [x] 리뷰 코멘트 공개 여부 변경 기능 문서화
- [x] 내가 받은 카드 및 리뷰 조회 기능 구현
- [x] 내가 받은 카드 및 리뷰 조회 기능 문서화  

## 📝리뷰어에게
이슈 128, 129를 한 PR로 올립니다.
내가 받은 카드 및 리뷰 조회 기능을 구현하면서 아래 형식처럼 `steady`별로 받은 리뷰를 묶어 주기 위해서 QueryDsl의 `transform`을 이용했습니다. 
  ```java
  {
    "reviews": [
          {
              "steadyId": 1,
              "steadyName": "블리츠",
              "reviews": [
                  {
                      "reviewId": 4,
                      "comment": "팀해줘서고마와용",
                      "isPublic": false,
                      "createdAt": "2023-11-14T17:59:04"
                  },  
                  {
                      "reviewId": 6,
                      "comment": "또 같이 공부하고 싶어요",
                      "isPublic": true,
                      "createdAt": "2023-11-14T17:59:45"
                  }
              ]
          }
      ]
  }
  ```
다만 이 때 `java.lang.NoSuchMethodError: 'java.lang.Object org.hibernate.ScrollableResults.get(int)'`라는 에러가 발생했고, 이에 대한 해결책으로 `QueryDslConfig.java`  코드를 수정했습니다.
- 관련 Commit: 80efa34c16535f2c747025471af9b1d2d872c482
- 관련 자료: [QueryDsl 이슈](https://github.com/querydsl/querydsl/issues/3428)

na-yk committed:
:white_check_mark: 리뷰 생성 기능 Controller 테스트 추가

na-yk committed:
:art: ReviewCreateRequest 패키지 이동

na-yk committed:
:art: JsonFieldType으로 변경

na-yk committed:
:sparkles: 리뷰 코멘트 공개여부 수정 기능 Service 추가

na-yk committed:
:sparkles: 리뷰 코멘트 공개여부 수정 기능 Controller 추가

na-yk committed:
:recycle: 리뷰 생성 기능 반환 URI 수정

na-yk committed:
:white_check_mark: 리뷰 코멘트 공개여부 수정 기능 Service 테스트 추가

na-yk committed:
:white_check_mark: ReviewFixture createReviewUpdateRequest 추가

na-yk committed:
:art: ReviewController updateReviewIsPublic 메서드명 오타 수정

na-yk committed:
:white_check_mark: 리뷰 코멘트 공개여부 수정 기능 Controller 테스트 추가

na-yk committed:
:art: Review 도메인 QueryRepository 메서드명 find->get 변경

na-yk committed:
:sparkles: 내 리뷰 조회 기능 Repository 추가

na-yk committed:
:sparkles: 내 리뷰 조회 기능 Service 추가

na-yk committed:
:sparkles: 내 리뷰 조회 기능 Controller 추가

na-yk committed:
:wrench: QueryDslConfig JPQLTemplates 지정

QueryDsl transform에서 발생하는 에러를 해결하기 위함

na-yk committed:
:white_check_mark: 내 리뷰 조회 기능 Service Test 추가

na-yk committed:
:art: ReviewMyResponse cards->userCards 변수명 변경

na-yk committed:
:white_check_mark: 내 리뷰 조회 기능 Controller Test 추가

na-yk committed:
:art: 컨벤션에 따라 코드 포맷팅

weonest commented on src/main/java/dev/steady/review/infrastructure/ReviewQueryRepositoryImpl.java:
```diff
@@ -28,6 +34,28 @@ public List<String> findPublicCommentsByRevieweeUser(User user) {
                 .fetch();
     }
 
+    @Override
+    public List<ReviewsBySteadyResponse> getAllReviewsByRevieweeUser(User user) {
+        return jpaQueryFactory.selectFrom(steady)
+                .leftJoin(review)
+                .on(steady.id.eq(review.steady.id))
+                .where(revieweeEqualsUser(user))
+                .orderBy(steady.finishedAt.desc())
+                .transform(groupBy(steady.id)
+                        .list(Projections.constructor(
+                                ReviewsBySteadyResponse.class,
+                                steady.id,
+                                steady.name,
+                                list(Projections.constructor(
+                                        ReviewDetailResponse.class,
+                                        review.id,
+                                        review.comment,
+                                        review.isPublic,
+                                        review.createdAt
+                                ))
+                        )));
+    }
```
프로젝션과 트랜스폼을 통한 한 번에 가져오는 로직을 너무 잘 짜신 것 같습니다 굿굿!

weonest commented on src/main/java/dev/steady/review/service/ReviewService.java:
```diff
@@ -75,6 +85,24 @@ public void createUserCards(ReviewCreateRequest request) {
+    @Transactional
+    public void updateReviewIsPublic(Long reviewId, ReviewUpdateRequest request, UserInfo userInfo) {
+        Review review = reviewRepository.getById(reviewId);
+        Long revieweeUserId = review.getReviewee().getUser().getId();
```
메서드 체인을 줄이면 좋을 것 같아요.

na-yk replied:
반영해보았습니다!

02f5feeb1401bbfb3c31779ca8a16b04bcf83ef3

weonest reviewed (approved):
너무 잘 짜신 것 같습니다. QeuryDSL 활용 능력에 감탄했어요 굿굿

na-yk committed:
:recycle: updateReviewIsPublic 메서드 체인 개선

na-yk committed:
:art: ReviewUpdateRequest 파일 제거

na-yk committed:
:sparkle: ReviewSwitchResponse 파일 추가

리뷰 공개 여부 변경 결과를 담는 응답 DTO

na-yk committed:
:sparkle: 리뷰 공개 여부 변경 기능 개선

기존에는 공개 여부 boolean 값을 요청으로 받아 해당 값으로 설정하였음. 요청에서 특정 값을 받지 않고 현재 저장된 값의 반대로 전환한 후, 그 결과를 응답으로 보내도록 개선함.
